### DISCUSSION TITLE
Issues 10/09/22

### DISCUSSION THREAD

**Discussion by andreievg** [msupply-foundation/open-msupply#111] [General]
## 10/09/22

Wanted to have a good grasp of issues with overview over all of the issues and their status on a regular basis.
This requires regular effort of looking through all of the issues (or previous issues reports) and updating their status and perhaps overall progress, comments and direction.

Trying to replicate something that @mark-prins was doing here https://github.com/openmsupply/org-issues/discussions/39
And [something](https://github.com/openmsupply/remote-server/issues/998) that I've started before holiday
And weekly HSH progress report emails compiled by @jmbrunskill and @Chris-Petty

Not really sure what the format should be like, but I think we'll figure it, the main thing is consistent and regular effort toward this task.

EDIT: just realised there is also summary of issues or requirements in: https://github.com/openmsupply/open-msupply/issues

**Comment by andreievg** [comment]
## [Remote Server (deprecated)](https://github.com/openmsupply/remote-server)

No more issues will be added but here is the summary of outstanding ones. They'll be relocated to open-msupply repo and removed when time to implement (or move to workload)

#### [V1 Internal Documentation](https://github.com/openmsupply/remote-server/issues/993)

This includes database schema documentation via SchemaSpy, current API pattern, back end layers and directory structure, general architecture and potential "how to", and rust docs

And all of these published to github pages via github hook (and then https://docs.msupply.foundation as suggested by @craigdrown)

- https://github.com/openmsupply/remote-server/issues/504
- https://github.com/openmsupply/remote-server/issues/230
- https://github.com/openmsupply/remote-server/issues/206

#### [V1 Improvements](https://github.com/openmsupply/remote-server/issues/994)

This was carried over from refactor work. Included using map_err vs from trait in services, ~removing invoice_id from line mutations~, re-using service within service and fetching invoice_lines and stock_lines in single query

- https://github.com/openmsupply/remote-server/issues/762
- ~~https://github.com/openmsupply/remote-server/issues/460~~ (@clemens-msupply commented this is not important and can be removed)

#### [Ungrouped V1 issues](https://github.com/openmsupply/remote-server/issues/782)

- https://github.com/openmsupply/remote-server/issues/482 and https://github.com/openmsupply/remote-server/issues/671
- use NonZeroU32 instead of checking manually
- https://github.com/openmsupply/remote-server/issues/487 (this has re appeared as a question in Finalise Sync V1 {add issues}, to know if we can continue skipping pulled records or stop operations due to sync error)
- https://github.com/openmsupply/remote-server/issues/560
- Able to create shipment from requisition by line (opional array added to mutation)

#### [Item and Name Details](https://github.com/openmsupply/remote-server/issues/1003)

- https://github.com/openmsupply/remote-server/issues/595 i think this was resolved with https://github.com/openmsupply/remote-server/pull/1100
- https://github.com/openmsupply/remote-server/issues/513 although we've added more details to item it's still in `legacy_record` json field on item

#### [Ungrouped Issues](https://github.com/openmsupply/remote-server/issues/783)

Issues without an epic with lower priority, not likely in V1

- ~~https://github.com/openmsupply/remote-server/issues/1002 (this is probably not low priority, tests were removed since they were using graphql client and were hard to read, they need to be reinstated)~~
- https://github.com/openmsupply/remote-server/issues/1001 (would be nice for item view to show similar charts as requisition line)
- https://github.com/openmsupply/remote-server/issues/999 (would be great, would need to be placeholder lines, but I don't think we can have quantity 0 placeholder line currently ?)
- https://github.com/openmsupply/remote-server/issues/982 (for one of the tabs in item view, although we can do this with master list resolver on item?)
- https://github.com/openmsupply/remote-server/issues/972 (can do with aggregation/view, some problems with date dimension though)
- https://github.com/openmsupply/remote-server/issues/943 (can specify or/and in query, pretty hard one as far as i can see)
- https://github.com/openmsupply/remote-server/issues/916 (some more info on master list)
- https://github.com/openmsupply/remote-server/issues/880 (for things like invad, to avoid conflicts, although it's central data so already should be free of conflicts)
- https://github.com/openmsupply/remote-server/issues/839 (currently we would mass delete lines for requisition or invoice if it's deleted, maybe that should not be allowed)
- https://github.com/openmsupply/remote-server/issues/811 to indicate that SOH snapshotIsHistoric, may also need some extras work on requisitions from https://docs.google.com/document/d/1AClFh5ZYv-X4PxQ86XBEj1LqMAM62CJRZbrt32qVYPk/edit
- https://github.com/openmsupply/remote-server/issues/606 (currently using availableBatches on item which is more cumbersome to use then itemsWithStockLines
- https://github.com/openmsupply/remote-server/issues/594 and https://github.com/openmsupply/remote-server/issues/593 and https://github.com/openmsupply/remote-server/issues/592 (extra info on invoice line node)
- https://github.com/openmsupply/remote-server/issues/551 (for fields that are not in sql query and are resolvers, would be hard due to pagination and sorting)

#### [Handle Donors in shipments](https://github.com/openmsupply/remote-server/issues/584)

Query, assign for invoice and lines lines

- https://github.com/openmsupply/remote-server/issues/583 and https://github.com/openmsupply/remote-server/issues/582 and https://github.com/openmsupply/remote-server/issues/581 and https://github.com/openmsupply/remote-server/issues/580

#### [Invoice Categories](https://github.com/openmsupply/remote-server/issues/784)

Query and assign to invoice

- https://github.com/openmsupply/remote-server/issues/587 and https://github.com/openmsupply/remote-server/issues/586 and https://github.com/openmsupply/remote-server/issues/585

#### [Shipping Methods](https://github.com/openmsupply/remote-server/issues/785)

Query and assign to invoice

- https://github.com/openmsupply/remote-server/issues/589 and https://github.com/openmsupply/remote-server/issues/588

#### [Transfers ](https://github.com/openmsupply/remote-server/issues/706)

Although transfers are already implemented, some extra bits that need to be done (using change log for sync processing, outline backwards compatibility with non v5 sites)

#### [Users and Permissions](https://github.com/openmsupply/remote-server/issues/786)

Just some carry over, as per issues:

- ~~https://github.com/openmsupply/remote-server/issues/793~~ (I think this is done now we are checking that store_id is for current site)
- https://github.com/openmsupply/remote-server/issues/936 (minor improvements)
- https://github.com/openmsupply/remote-server/issues/745 (should ignore case and existing msupply shows usernames, but this is bad for security for cloud remote, but can be done for local remote)

#### [Customer Credits](https://github.com/openmsupply/remote-server/issues/763)

This still needs full spec, can be done now I think with outbound shipments back to supplier

#### [Supplier Returns](https://github.com/openmsupply/remote-server/issues/765)

Similar to Customer Credits

#### [Deployment Ready ](https://github.com/openmsupply/remote-server/issues/788)

I think only carry over is Installer (which has seperate issue in open-msupply repository

#### [Sync after V1 beta](https://github.com/openmsupply/remote-server/issues/816)

There are extras to be added here, will be summed up in open-msupply repo issue, but in remote server repo. Need to sum this up in more details

- https://github.com/openmsupply/remote-server/issues/791 (this requires central pull to show deletes, this needs a more spec out, for central store specific data too, i.e. name_store_join)
- https://github.com/openmsupply/open-msupply/issues/69 (this also has an issue in open-msupply repo)

#### [Stocktake Remainder](https://github.com/openmsupply/remote-server/issues/997)

- https://github.com/openmsupply/remote-server/issues/924 (we are using just one inventory adjustment currently, but can have multiple inventory adjustments coming from legacy)
- https://github.com/openmsupply/remote-server/issues/895 (need an agreement, for cases when snapshot quantity has changed due to invoices created prior to stocktake or after stocktake started, here is a conversation i dug up: https://github.com/openmsupply/mobile/issues/638)

#### [Pricing Alignment](https://github.com/openmsupply/remote-server/issues/947)

This is somewhat important, we doing different things now in outbound and inbound, also final agreed solution is slightly different from both: https://github.com/openmsupply/remote-server/issues/947

#### Report

* https://github.com/openmsupply/remote-server/issues/980 (default query questions/requirements and where they are stored)
* https://github.com/openmsupply/remote-server/issues/954 (allow self cert when self query API for report data)


#### [Requisitions Remainder](https://github.com/openmsupply/remote-server/issues/995)

Ability to create shipments from requisition lines (this is already mentioned in ungrouped v1) and refreshing soh and amc via api call, also mentioned in ungrouped issues https://github.com/openmsupply/remote-server/issues/811 (clarify that snapshot quantity is history)

#### [Logging](https://github.com/openmsupply/remote-server/issues/1000)

This one is for logging server/user action but I think an overarching epic would be better to also combine https://github.com/openmsupply/open-msupply/issues/23 and needs to consider logging of events for debugging error state when we deploy

#### [Test Image](https://github.com/openmsupply/open-msupply/issues/23)

This was for docker mainly, it's done in: https://github.com/openmsupply/omsupply-build-tools, it may need some updates if there is interest in using it (if there is then it should be relocated to open-msupply repo, otherwise removed).

#### Other ungrouped

Don't have an epic

[~~Deployment ready~~](https://github.com/openmsupply/remote-server/issues/1016) this was an attempt to sum up what's needed for deployment with some discussion about SSL, we've solved most of this with desktop electron bundle and only really installer remains.

[~~Response Requisition: remainingQuantityToSupply is always 0~~](https://github.com/openmsupply/remote-server/issues/1017) I think it's been solved, to confirm

[Modified by & modified datetime](https://github.com/openmsupply/remote-server/issues/1018) Comes from a request to show last modified by and last modified in UI

[Derive report data query permissions](https://github.com/openmsupply/remote-server/issues/1034) Looks like for UI to show greyed out report (quick note, do they need to be greyed out ?)

[Postman collection](https://github.com/openmsupply/remote-server/issues/1043) Should be part of docs section

[Sync settings: test](https://github.com/openmsupply/remote-server/issues/1046) When initialising check settings and report back to UI (it's covered in other repo https://github.com/openmsupply/open-msupply/issues/66)

[Review design for item store relationship](https://github.com/openmsupply/remote-server/issues/1053) Some discussion about new way of doing item store relationship, I think it's resolved, but can keep as open issue

[~~Failing to pull all items from Central server~~](https://github.com/openmsupply/remote-server/issues/1095) This is resolved

[Extra login protection when user password has changed](https://github.com/openmsupply/remote-server/issues/1101) Although immediate issue was resolved, need to consider one tricky use case

[~~Authentication Error upon logging](https://github.com/openmsupply/remote-server/issues/1108~~) Resolved

**Reply by Chris-Petty** [reply]
If it helps, moving issues from one repo to another should be fairly easy:

![Screen Shot 2022-06-10 at 10 09 49 AM](https://user-images.githubusercontent.com/7684221/172953834-d8fe93cb-030e-42c2-b334-0d020c485ccd.png)
_(right sidebar on issue)_

But still 1 by 1! Not sure if there is a bulk way... wonder if it could be done through github gql api... probably faster to do 1 by 1 but less interesting haha

**Reply by jmbrunskill** [reply]
https://jloh.co/posts/bulk-migrate-issues-github-cli/

**Comment by andreievg** [comment]
## [Open mSupply](https://github.com/openmsupply/open-msupply/issues/68)

#### [Report PDF Printing](https://github.com/openmsupply/open-msupply/issues/68)

I am not sure exactly what's outstanding apart from issues linked in remote server (see `Report` section) and Android printing:

* https://github.com/openmsupply/open-msupply/issues/13 this is fixed in a PR but some issues with self-signed cert in browser and exploring `can the header and footer show on every page`
* https://github.com/openmsupply/open-msupply/issues/71 (allow referencing resource from other reports)

#### Programs

* https://github.com/openmsupply/open-msupply/issues/1
* https://github.com/openmsupply/open-msupply/issues/7
* https://github.com/openmsupply/open-msupply/issues/8
* https://github.com/openmsupply/open-msupply/issues/25
* https://github.com/openmsupply/open-msupply/issues/27
* https://github.com/openmsupply/open-msupply/issues/29
* https://github.com/openmsupply/open-msupply/issues/30
* https://github.com/openmsupply/open-msupply/issues/31
* https://github.com/openmsupply/open-msupply/issues/32
* https://github.com/openmsupply/open-msupply/issues/33
* https://github.com/openmsupply/open-msupply/issues/99
* https://github.com/openmsupply/open-msupply/issues/74 (some issues from PR)
* https://github.com/openmsupply/open-msupply/issues/76
* https://github.com/openmsupply/open-msupply/issues/81 
* https://github.com/openmsupply/open-msupply/issues/101
* https://github.com/openmsupply/open-msupply/issues/103

There is also discussion about sync and merge (I think we are leaning towards simplicity, and potentially removing merging, hash, multiple parents and sacrificing by field change record as it's very rare scenario) https://github.com/openmsupply/open-msupply/issues/88

#### Ungrouped

These are issues without epic. Would be good to discuss if we want to continue with epic labels from remote server (some of these are epics themselves)

[Number table rows requisitions](https://github.com/openmsupply/open-msupply/issues/69) Currently using response and request requisition (mSupply just has requisition I think and it's for both)

[Permissions via API](https://github.com/openmsupply/open-msupply/issues/67) Query for permissions, this also mentions more granular report permissions with linked remote server issue [Report PDF Printing](https://github.com/openmsupply/open-msupply/issues/68), but talks about redefining current resources

[Bootstrap mode open issues (Initialisation state, sync status, manual sync](https://github.com/openmsupply/open-msupply/issues/66), started of as a carry over from initialisation/bootstrap mode, but a comment added listing F/E requirement for sync status, manual sync and possibly removing bootstrap mode (or possibly changing it).

[Sqlite build config on Android](https://github.com/openmsupply/open-msupply/issues/70) I think we have a temporary fix but need a more permanent fix for cache directory for sqlite to use

[Audit log](https://github.com/openmsupply/open-msupply/issues/23) Logging use activity, also it's mentioned in `Logging` of remote server issue summary. I think we should also brainstorm logging of activities for debugging in this issue (or at least create a new issue for it)

[Sync Issues](https://github.com/openmsupply/open-msupply/issues/21) ~Include hardware id in headers,~ extra work for V5 to work. See Remote Server `Sync after V1 beta`. Something that hasn't been added to the issue but should be (from chat summary with Chris):
* Check referential integrity on large data files (based on referential constraints on remote server)
  * Ok -> error out on remote server pull
  * Not -> can skip errors (record them of course, and perhaps try to re-integrate them nek time, i.e. keep the sync buffer)
* Strategy for buffer on error or on skip and how to ‘save’ it for migrations

There is also fractional number of packs, with discussions here https://github.com/openmsupply/org-issues/discussions/40 to me the summary of it is: much simpler to implement fractional number of packs (I think the main origin of it is placeholder line distribution, so this area should also be redistributed)

[Global constants file](https://github.com/openmsupply/open-msupply/issues/42) for front end constants like records per page

[Dependency update](https://github.com/openmsupply/open-msupply/issues/52) updating front end dependencies to latest version

[Electron client improvements](https://github.com/openmsupply/open-msupply/issues/58) Mark has done most of them there are some extras
* Make it secure (ask user to trust cert)
* Reduce CPU usage when discovering
* Properly stop and start discovery
* Proper mac window behaviour
* Potential other libraries for server mDNS
* Generate self signed cert on startup (or when installed)

[Reinstate running restrictive cors in dev](https://github.com/openmsupply/open-msupply/issues/60) Reinstate flag to still run cors in dev and cors tests

[Diesel async](https://github.com/openmsupply/open-msupply/issues/78) Transition to use async everywhere, should accompany benchmark (or maybe start with benchmark)

[Remove sqlite enum constraints](https://github.com/openmsupply/open-msupply/issues/82) should be small, just remove `with check` form sqlite, we have enums in PG don't need to define in two places

[Sqlite can panic while creating a new database and WAL on startup](https://github.com/openmsupply/open-msupply/issues/84) Sometimes sqlite panics creating new db and wall on startup

[~~Incorrect info output during sync~~](https://github.com/openmsupply/open-msupply/issues/97) Resolved

[Compile error due to Autocomplete types](https://github.com/openmsupply/open-msupply/issues/104) Front end TS error for Autocomplete package

[~~ Incorrect authentication error notification (f/e) ~~](https://github.com/openmsupply/open-msupply/issues/109) Resolved, but still some discussions about persisting token bucket after server restart

**Comment by andreievg** [comment]
## [Open mSupply Client (deprecated)](https://github.com/openmsupply/openmsupply-client)
